pkg/dns/dns.go

```diff
@@ -37,10 +37,7 @@ func GetMultiServiceFQDN(stsName string, namespace string, clusterNum int, podNu
 		domain = strings.TrimPrefix(clusterDomain, ".")
 	}
 
-	// For StatefulSet pods, DNS format is: <pod-name>.<headless-service>.<namespace>.svc.<domain>
-	podName := GetMultiPodName(stsName, clusterNum, podNum)
-	headlessService := GetMultiHeadlessServiceName(stsName, clusterNum)
-	return fmt.Sprintf("%s.%s.%s.svc.%s", podName, headlessService, namespace, domain)
+	return fmt.Sprintf("%s.%s.svc.%s", GetMultiServiceName(stsName, clusterNum, podNum), namespace, domain)
```

Comment:
Reverting back to this (original, as seen in master) code fixes most of the tests, but breaks `MongoDB`-based multi-cluster.

Controller for `MongoDBMultiCluster` generates hostnames like this:

```
multi-replica-set-0-0-svc.mongodb-test.svc.cluster.local
multi-replica-set-1-0-svc.mongodb-test.svc.cluster.local
multi-replica-set-2-0-svc.mongodb-test.svc.cluster.local
```

But the controller for `MongoDB` multi-cluster replica sets creates hostnames like this even with the revert:

```
multi-replica-set-0-0.multi-replica-set-0-svc.mongodb-test.svc.cluster.local
multi-replica-set-1-0.multi-replica-set-1-svc.mongodb-test.svc.cluster.local
multi-replica-set-2-0.multi-replica-set-2-svc.mongodb-test.svc.cluster.local
```

Still trying to understand why.